### ISSUE TITLE
Having an issue decoding an early transaction

### ISSUE THREAD

**Issue by cryptoquick** [rust-bitcoin/rust-bitcoin#3177]
I'm debugging this code:

https://github.com/SurmountSystems/gabriel/blob/8362fdcfd686e4ec461286f4ed0463da747aaf83/src/main.rs#L79-L101

Output is:

```
https://xyz.local/api/block-height/540
https://xyz.local/api/block/000000001958fb614e3eed2ab35305fa89a8b8db9bb5fd2575aa08fc0a06b84f
tx len: 1
https://xyz.local/api/block-height/541
https://xyz.local/api/block/00000000aa3704d9c763fac9ae41842c55a2b3a1c5b404dff2ca82c9f9ebd64b
tx len: 1
https://xyz.local/api/block-height/542
https://xyz.local/api/block/00000000bdb2a11dbc79d6f367e666448ac16aed9be8ad7c5a5712e23d0a4942
tx len: 1
https://xyz.local/api/block-height/543
https://xyz.local/api/block/0000000009b905f3db4990848b9b14fe39582147fcf6a6cb8d73bd227d75369e
tx len: 1
https://xyz.local/api/block-height/544
https://xyz.local/api/block/000000002d947997dc957cdf075dd32390f5f754d2656208d5dd82a6620179f5
tx len: 1
https://xyz.local/api/block-height/545
https://xyz.local/api/block/00000000689051c09ff2cd091cc4c22c10b965eb8db3ad5f032621cc36626175
tx len: 1
tx len: 1
https://xyz.local/api/block-height/546
https://xyz.local/api/block/000000005a4ded781e667e06ceefafb71410b511fe0d5adc3e5a27ecbec34ae6
tx len: 1
tx len: 2
tx len: 2
res len: 217
hex: 3c21444f43545950452068746d6c3e0a3c68746d6c206c616e673d22656e223e0a3c686561643e0a3c6d65746120636861727365743d227574662d38223e0a3c7469746c653e4572726f723c2f7469746c653e0a3c2f686561643e0a3c626f64793e0a3c7072653e43616e6e6f7420474554202f6170692f76312f74782f323832303463616431643766633164313939653865663466613232663138326465363235386133656161666531626265353665626463616364333036396135662f7261773c2f7072653e0a3c2f626f64793e0a3c2f68746d6c3e0a
real len: 217
test len: 193
Error: Io(Error { kind: UnexpectedEof, error: None })
```

Sorry if this issue is short... It should be obvious what I'm trying to accomplish, but if it isn't, feel free to ask for clarification.

**Comment by apoelstra** [comment]
The hex you posted does not look like a transaction. It does not decode with Bitcoin Core either.

**Comment by cryptoquick** [comment]
Aha! You are correct. That's because the hex I provided is actually an encoded version of this response text:

```
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /api/v1/tx/cb6cae7d02a30fa4e6fc2d9ff83dc8acae967657e187d1ffc3caca97e28785cd/raw</pre>
</body>
</html>
```

I'll have to figure out why I can't get the transaction by txid...

Is there a better way to get the spend scripts used by the inputs?

**Comment by cryptoquick** [comment]
I think there's some kind of bug in the mempool version I'm using that precludes raw transaction output. I had to implement my own serde structs for its own particular JSON format.

I'm going to close this now as I think this is growing out of scope. Thank you for helping me get unblocked.
